v1/tarball/image_test.go

```diff
@@ -15,11 +15,70 @@
 package tarball
 
 import (
+	"archive/tar"
+	"io"
+	"os"
+	"path/filepath"
 	"testing"
 
 	"github.com/google/go-containerregistry/name"
 )
 
+func TestFlatten(t *testing.T) {
+	img, err := ImageFromPath("testdata/whiteout.tar", nil)
+	if err != nil {
+		t.Errorf("Error loading image: %v", err)
+	}
+	tarPath, _ := filepath.Abs("img.tar")
+	defer os.Remove(tarPath)
+	if err := Flatten(img, tarPath); err != nil {
+		t.Errorf("Error when flattening image: %v", err)
+	}
+	f, err := os.Open(tarPath)
+	if err != nil {
+		t.Errorf("Error when opening tar file for reading: %v", err)
+	}
+	tr := tar.NewReader(f)
+	for {
+		header, err := tr.Next()
+		if err == io.EOF {
+			break
+		}
+		name := header.Name
+		// this image was built by creating a directory called "foo",
+		// touching "/foo/bar", and then removing the whole directory.
+		for _, part := range filepath.SplitList(name) {
```

Comment:
We should also add a file to the tar that exercises a file being overwritten in a subsequent layer.